src/main/java/mat/client/Login.java

```diff
@@ -58,127 +57,81 @@ public void onSuccess(final Void arg0) {
-				content.clear();
-				loginNewPresenter.go(content);
-				loginNewPresenter.displayForgottenPasswordMessage();
-			}
+		MatContext.get().getEventBus().addHandler(PasswordEmailSentEvent.TYPE, event -> {
+			content.clear();
+			loginNewPresenter.go(content);
+			loginNewPresenter.displayForgottenPasswordMessage();
 		});
-		
-		MatContext.get().getEventBus().addHandler(ForgotLoginIDEmailSentEvent.TYPE, new ForgotLoginIDEmailSentEvent.Handler() {
-			
-			@Override
-			public void onForgotLoginIdEmailSent(final ForgotLoginIDEmailSentEvent event) {
-				content.clear();
-				 loginNewPresenter.go(content);
-				 loginNewPresenter.displayForgottenLoginIDMessage();
-			}
-		});
-		MatContext.get().getEventBus().addHandler(ForgottenPasswordEvent.TYPE, new ForgottenPasswordEvent.Handler() {
-			
-			@Override
-			public void onForgottenPassword(final ForgottenPasswordEvent event) {
-				content.clear();
-				forgottenPwdPresenter.go(content);
-			}
-		});
-		
-		MatContext.get().getEventBus().addHandler(ForgotLoginIDEvent.TYPE, new ForgotLoginIDEvent.Handler() {
-			
-			@Override
-			public void onForgottenLoginID(final ForgotLoginIDEvent event) {
-				content.clear();
-				forgottenLoginIdNewPresenter.go(content);
-			}
+
+		MatContext.get().getEventBus().addHandler(ForgotLoginIDEmailSentEvent.TYPE, event -> {
+			content.clear();
+			 loginNewPresenter.go(content);
+			 loginNewPresenter.displayForgottenLoginIDMessage();
 		});
-		
-		MatContext.get().getEventBus().addHandler(SuccessfulLoginEvent.TYPE, new SuccessfulLoginEvent.Handler() {
-			
-			@Override
-			public void onSuccessfulLogin(final SuccessfulLoginEvent event) {
-				SecurityBannerModal securityBanner = new SecurityBannerModal();
-				securityBanner.show();
-				securityBanner.getAcceptButton().addClickHandler(e -> {
-					securityBanner.hide();
-					MatContext.get().redirectToHtmlPage(ClientConstants.HTML_MAT);
-				});
-				securityBanner.getDeclineButton().addClickHandler(e -> {
-					securityBanner.hide();
-					callSignOut();
-				});
-			}
+		MatContext.get().getEventBus().addHandler(ForgottenPasswordEvent.TYPE, event -> {
+			content.clear();
+			forgottenPwdPresenter.go(content);
 		});
-		
-		MatContext.get().getEventBus().addHandler(ReturnToLoginEvent.TYPE, new ReturnToLoginEvent.Handler() {
-			
-			@Override
-			public void onReturnToLogin(final ReturnToLoginEvent event) {
-				content.clear();
-				loginNewPresenter.go(content);
-			}
+
+		MatContext.get().getEventBus().addHandler(ForgotLoginIDEvent.TYPE, event -> {
+			content.clear();
+			forgottenLoginIdNewPresenter.go(content);
 		});
-		
-		MatContext.get().getEventBus().addHandler(BackToLoginPageEvent.TYPE, new BackToLoginPageEvent.Handler() {
-			
-			@Override
-			public void onLoginFailure(final BackToLoginPageEvent event) {
-				MatContext.get().redirectToHtmlPage(ClientConstants.HTML_LOGIN);
-			}
+
+		MatContext.get().getEventBus().addHandler(SuccessfulLoginEvent.TYPE, event -> {

```

Comment:
Does this actually work? I recall we had issues in similar cases. Java closure event handlers were not working properly and I had to rollback to 'Anonymous Inner Classes'..

Reply:
they are used all over mat gwt code now.